type_description_interfaces/msg/TypeSource.msg

```diff
@@ -0,0 +1,16 @@
+# Represents the original source of a ROS 2 interface definition.
+
+uint8 ENCODING_NOT_SET = 0
+uint8 ENCODING_MSG = 1
+uint8 ENCODING_SRV = 2
+uint8 ENCODING_ACTION = 3
+uint8 ENCODING_IDL = 4
+
+# ROS interface type name, in PACKAGE/NAMESPACE/TYPENAME format
+string type_name
+
+# The type of the original source file, from the above constants
+uint8 encoding
```

Comment:
I'm not sure about the use of an enum for this. Makes it hard to expand in the future. I was thinking we'd use a string, and for now default to using the file extension as the identifiers.

Imagine the process for using protobuf here.

Reply:
Yes, I see your point - changing to a string and listing the well-known values